### ISSUE TITLE
Access Denied for SSH showing unique id, not nodename

### ISSUE THREAD

**Issue by stevenGravy** [gravitational/teleport#23719]
Expected behavior:

The access denied shows the hostname, not the uuid.

Current behavior:

The access denied is showing the node unique id.

<img width="299" alt="image" src="https://user-images.githubusercontent.com/60704961/228350748-c237f5ce-7b84-4e75-b8b9-a80bed9aac57.png">

Bug details:
- Teleport version: 12.1.2
- Recreation steps

Connect to a ssh node with the wrong username in Web UI or Teleport Connect.

**Comment by stevenGravy** [comment]
This has started occurring again. If the user puts in a non-allowed user they recieve this message with the node uuid.

![Image](https://github.com/user-attachments/assets/fc7d0bc5-4758-435b-86d0-23f335a3f8b1)

**Comment by zmb3** [comment]
> @zmb3 is there any reason for the Web UI to allow users to input a custom login? Shouldn't all of the allowed logins be available in the dropdown list when clicking the connect button? Instead of allowing freeform text to be passed in for the login should that text input be converted to a filter for the available logins?

You're correct. @avatus is actually working on removing the free form login entry for v17 right now.

**Comment by avatus** [comment]
https://github.com/gravitational/teleport/pull/47422
